TSPL.docc/LanguageGuide/Concurrency.md

```diff
@@ -564,33 +637,192 @@ the explicit parent-child relationships between tasks
 await withTaskGroup(of: Data.self) { taskGroup in
+    let photoNames = await listPhotos(inGallery: "Summer Vacation")
+    for name in photoNames {
+        taskGroup.addTask {
+            let photo = downloadPhoto(named: name)
+            show(photo)
+        }
+    }
+}
+```
+
+The code above creates a new task group,
+and then creates child tasks of that task group
+to download each photo is the gallery.
+Each task group downloads a single photograph
+and then adds it to the photographs being displayed.
+Because there isn't a specific priority passed
+when calling `addTask(priority:operation:)`,
+these child tasks tasks get the same priority as the task group.
+Swift schedules these tasks,
+running as many of them at the same time as conditions allow.
+
+> Note:
+> If the code to download a photo could throw an error,
+> you would call `withThrowingTaskGroup(of:returning:body:)` instead.
+
+In the code listing above,
+each photo is downloaded and immediately displayed,
+so the task group doesn't return any results.
+For a task group that returns a result,
+you add code that accumulates its result
+inside the closure you pass to `withTaskGroup(of:returning:body:)`.
+
+```
+let photos = await withTaskGroup(of: Data.self) { taskGroup in
     let photoNames = await listPhotos(inGallery: "Summer Vacation")
     for name in photoNames {
         taskGroup.addTask { await downloadPhoto(named: name) }
     }
+
+    var results: [Data] = []
+    for await photo in taskGroup {
+        results.append(photo)
+    }
+    return results
 }
 ```
 
-<!--
-  TODO walk through the example
+Like the previous example,
+this example creates a child task for each photo to download it.
+Unlike the previous example,
+the `for`-`await`-`in` loop waits for the next child task to finish,
+appends the result of that task to the array of results,
+and then continues waiting until all child tasks have finished.
+Finally,
+the task group returns the array of downloaded photos
+as its overall result.
+
+
+### Task Cancellation
+
+Swift concurrency uses a cooperative cancellation model.
+Each task checks whether it has been canceled
+at the appropriate points in its execution,
+and responds to cancellation an appropriate way.
+Depending on the work you're doing,
+that usually means one of the following:
+
+- Throwing an error like `CancellationError`
+- Returning `nil` or an empty collection
+- Returning the partially completed work
+
+Downloading pictures could take a long time
+if the pictures are large or the network is slow.
+To let the user stop this work,
+without waiting for all of the tasks to complete,
+the tasks need check for cancellation and stop running if they are canceled.
+There are two ways a task can do this:
+by calling the [`Task.checkCancellation()`][] method,
+or by reading the [`Task.isCancelled`][] property.
+Calling `checkCancellation()` throws an error if the task is canceled;
+a throwing task can let that error propagate out of the task,
```

Comment:
This reads weirdly to me. Maybe something more like "... a throwing task can propagate the error out of the task, ...`?

Reply:
Changed.

I think this was just my personal tendency to think of error propagation as something you passively "let" happen — because you didn't write any code to actively/explicitly propagate the error.  Rephrased here, to match the way TSPL talks about propagation in the error handling chapter.